presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/IndexJoinOptimizer.java

```diff
@@ -106,6 +106,10 @@ public PlanOptimizerResult optimize(
             rewriter = new DefaultJoinRewriter(idAllocator, metadata, session);
         }
         PlanNode rewrittenPlan = SimplePlanRewriter.rewriteWith(rewriter, plan, null);
+        if (SystemSessionProperties.isNativeExecutionEnabled(session) && !rewriter.isPlanChanged()) {
```

Comment:
can you explain the bug this is fixing?  what does the rewrittenPlan look like if rewriter.isPlanChanged is false? Shouldn't it be the same as the original plan?

Reply:
IndexJoin Java only supports equal join condition (=). We extended this optimizer to support non-equal join condition (BETWEEN, CONTAINS), which is expressed as an additional FilterNode on top of the Join. In order to make the conditions visible when considering turn the Join into IndexJoin, this optimizer pushes the FilterNode into the JoinNode hence plan changed. But we should keep the two nodes separate if index join doesn't happen.